app/src/forms/form/controller.js

```diff
@@ -387,4 +387,52 @@ module.exports = {
       next(error);
     }
   },
+  listExternalAPIs: async (req, res, next) => {
```

Comment:
One of my responsibilities this quarter is to make sure that maintainability of the backend improves (we currently rate a `D`). This file (and many others) are flagged by [Code Climate](https://codeclimate.com/github/bcgov/common-hosted-form-service/app/src/forms/form/controller.js) as having too many functions (or too many lines, etc). We're never going to fix complaints about similar code where a `delete` function calls `service.delete`, `update` calls `service.update`, etc, but the size of some of the files is getting awkward. The way Code Climate is set up we don't really have a way of seeing problems until merges to main happen - so it's entirely expected that new code will follow the old pattern that Code Climate complains about! My way of working has been to check Code Climate for any files I change, and see if I can avoid adding to the list of issues.

Unfortunately I don't have a new pattern figured out yet that can be worked from, but it would be great if new code didn't add to the maintenance that will need to be done. Two very vague ideas I had were:
1. Have subdirectories for each "feature", like `forms/form/externalApi`, and that directory would have its own `route.js` (or not?) and `controller.js` and `service.js`. Preferred?
2. Have multiple files in the `forms/form` directory, like `routeExternalApi.js` (?) , `controllerExternalApi.js`, `serviceExternalApi.js`, etc.

If we could figure a pattern out, then eventually I'll start to break down the big controller and service files to match the pattern.

Reply:
let's come up with something and try it out. the folder approach is probably the one I'd try first.

Reply:
moved it into sub-folder as per suggestion 1.

Reply:
Looks great!